hack/release-from/action.go

```diff
@@ -0,0 +1,305 @@
+// Copyright (c) 2025 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v3"
+	"gopkg.in/yaml.v3"
+)
+
+// releaseFrom is the main action for the command.
+// It builds (and publishes) a new operator based on the base version specified.
+func releaseFrom(ctx context.Context, c *cli.Command) error {
+	// get root directory of operator git repo
+	repoRootDir, err := runCommand("git", []string{"rev-parse", "--show-toplevel"}, nil)
+	if err != nil {
+		return fmt.Errorf("Error getting git root directory: %s", err)
```

Comment:
For golang users such as myself, there are a lot of warnings. Maybe nice to clean up.
https://go.dev/wiki/CodeReviewComments#error-strings

> Error Strings[¶](https://go.dev/wiki/CodeReviewComments#error-strings)
> Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with > punctuation, since they are usually printed following other context. That is, use fmt.Errorf("something bad") not fmt.Errorf("Something bad"), so that log.Printf("Reading %s: %v", filename, err) formats without a spurious capital letter mid-message. This does not apply to logging, which is implicitly line-oriented and not combined inside other messages.